packages/ui5-task-pwa-enabler/lib/pwaEnabler.js

```diff
@@ -0,0 +1,286 @@
+    }
+    // Write the service worker
+    await renderServiceWorker(path.join(__dirname, "../templates/cacheFirst.js"), {preCache: fileList});
+    return undefined;
+}
+
+/**
+ * Takes the template for the advancedCache service worker and fills it with the values provided in parameters
+ *
+ * @param {object} parameters
+ * @param {string} parameters.preCache Path to the file that will be used as offline page
+ * @param {string} parameters.networkFirst Path to the file that will be used as offline page
+ * @param {string} parameters.avoidCaching Path to the file that will be used as offline page
+ * @param {string} parameters.offlinePage Path to the file that will be used as offline page
+ * @returns {Promise<undefined>}
+ */
+async function addAdvancedCachingSw(parameters) {
+    // First collect the pre cache files
+    // We need to prefix the pattern with our rootDir
+    let globPattern = [];
+    for (let i = 0; i < parameters.preCache.length; i++) {
+        let pattern = undefined;
+        // Check for leading slash before concatenating paths
+        if (parameters.preCache[i][0] === "/") {
+            pattern = rootDir + parameters.preCache[i];
+        } else {
+            pattern = rootDir + "/" + parameters.preCache[i];
+        }
+        globPattern.push(pattern);
+    }
+    // Get all resources that match the pattern and make them into a nice list
+    let resources = await ws.byGlob(globPattern);
+    let preCache = "";
+    for (let i = 0; i < resources.length; i++) {
+        // Remove prefix
+        let path = resources[i].getPath().replace(rootDir, "");
+        preCache = preCache + '\n"' + path + '",';
+    }
+    // Concatenate all regular expressions for networkFirst
+    let networkFirst = "";
+    for (let i = 0; i < parameters.networkFirst.length; i++) {
+        networkFirst = networkFirst + '\n"' + parameters.networkFirst[i] + '",';
+    }
+    // Concatenate all regular expressions for networkFirst
+    let avoidCaching = "";
+    for (let i = 0; i < parameters.avoidCaching.length; i++) {
+        avoidCaching = avoidCaching + '\n"' + parameters.avoidCaching[i] + '",';
+    }
+    // Write the service worker
+    await renderServiceWorker(path.join(__dirname, "../templates/advancedCache.js"), {
+        offlinePage: parameters.offlinePage,
+        preCache: preCache,
+        networkFirst: networkFirst,
+        avoidCaching: avoidCaching
+    });
+    return undefined;
+}
+
+/**
+ * Utility function to easily write the service worker
+ *
+ * @param {string} serviceWorkerTemplate path to the service worker template file
+ * @param {object} view will be passed to mustache.render to read values from it
+ * @returns {Promise<undefined>}
+ */
+async function renderServiceWorker(serviceWorkerTemplate, view) {
+    let path = rootDir + "/service-worker.js";
+    let swCode = fs.readFileSync(serviceWorkerTemplate);
+    view.timestamp = new Date().getTime().toString();
```

Comment:
you can also use `Date.now()` - it is a number - but you could convert it by adding it to a string